src/connector/src/sink/iceberg/mod.rs

```diff
@@ -938,6 +940,88 @@ enum ProjectIdxVec {
     Done(Vec<usize>),
 }
 
+type DataFileWriterBuilderType =
+    DataFileWriterBuilder<ParquetWriterBuilder, DefaultLocationGenerator, DefaultFileNameGenerator>;
+type PositionDeleteFileWriterBuilderType = PositionDeleteFileWriterBuilder<
+    ParquetWriterBuilder,
+    DefaultLocationGenerator,
+    DefaultFileNameGenerator,
+>;
+type EqualityDeleteFileWriterBuilderType = EqualityDeleteFileWriterBuilder<
+    ParquetWriterBuilder,
+    DefaultLocationGenerator,
+    DefaultFileNameGenerator,
+>;
+
+#[derive(Clone)]
+struct TaskWriterBuilderWrapper<B: IcebergWriterBuilder> {
+    inner: B,
+    fanout_enabled: bool,
+    schema: IcebergSchemaRef,
+    partition_spec: PartitionSpecRef,
+    compute_partition: bool,
+}
+
+impl<B: IcebergWriterBuilder> TaskWriterBuilderWrapper<B> {
+    fn new(
+        inner: B,
+        fanout_enabled: bool,
+        schema: IcebergSchemaRef,
+        partition_spec: PartitionSpecRef,
+        compute_partition: bool,
+    ) -> Self {
+        Self {
+            inner,
+            fanout_enabled,
+            schema,
+            partition_spec,
+            compute_partition,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl<B: IcebergWriterBuilder> IcebergWriterBuilder for TaskWriterBuilderWrapper<B> {
+    type R = TaskWriter<B>;
+
+    async fn build(self, partition_key: Option<PartitionKey>) -> iceberg::Result<Self::R> {
+        let partition_splitter = if self.partition_spec.is_unpartitioned() {
+            None
+        } else if self.compute_partition {
+            Some(RecordBatchPartitionSplitter::new_with_computed_values(
+                self.schema.clone(),
+                self.partition_spec.clone(),
+            )?)
+        } else {
+            Some(RecordBatchPartitionSplitter::new_with_precomputed_values(
+                self.schema.clone(),
+                self.partition_spec.clone(),
+            )?)
+        };
+
+        // For unpartitioned tables, ignore any provided partition key.
+        if self.partition_spec.is_unpartitioned() {
```

Comment:
This logic is really weird. We aren't using the **partition_key**, and we use the same piece of build code regardless of whether the partition_spec is empty or not.

could we simplify it?